### PR TITLE
 Throw descriptive error when filtering without condition - 7.3.x

### PR DESCRIPTION
**Pull request by tachojelev**
#4897

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [x] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [x] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/api.service.ts [hunk 1]
```diff
@@ -326,6 +326,9 @@ export class GridBaseAPIService <T extends IgxGridBaseComponent & IGridDataBinda
         }
 
         grid.filteringExpressionsTree = filteringTree;
+        if (!condition) {
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/grids/api.service.ts]: You could move this check at the beginning of the method. However I think that for filterGlobal, we should make the condition argument required. This method doesn't make sense to be called without condition. Both changes are breaking, but I think that compilation error is better than a runtime error in this case.

File: projects/igniteui-angular/src/lib/grids/api.service.ts [hunk 2]
```diff
@@ -310,6 +310,10 @@ export class GridBaseAPIService <T extends IgxGridBaseComponent & IGridDataBinda
     }
 
     public filter_global(term, condition, ignoreCase) {
+        if (!condition) {
+            throw Error('Invalid condition!');
```
Comment (bkulov) [inline projects/igniteui-angular/src/lib/grids/api.service.ts]: You should create an instance of the error here - ` throw new Error`.

File: projects/igniteui-angular/src/lib/grids/filtering/grid-filtering.service.ts [hunk 3]
```diff
@@ -150,7 +150,9 @@ export class IgxFilteringService implements OnDestroy {
             this.gridAPI.filter(field, value, conditionOrExpressionTree, filteringIgnoreCase);
         } else {
             const expressionsTreeForColumn = this.grid.filteringExpressionsTree.find(field);
-            if (expressionsTreeForColumn instanceof FilteringExpressionsTree) {
+            if (!expressionsTreeForColumn) {
+                throw Error('Invalid condition or Expression Tree!');
```
Comment (bkulov) [inline projects/igniteui-angular/src/lib/grids/filtering/grid-filtering.service.ts]: Same as above - ` throw new Error`.

File: projects/igniteui-angular/src/lib/grids/api.service.ts [hunk 4]
```diff
@@ -310,6 +310,10 @@ export class GridBaseAPIService <T extends IgxGridBaseComponent & IGridDataBinda
     }
 
     public filter_global(term, condition, ignoreCase) {
+        if (!condition) {
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/grids/api.service.ts]: It doesn't seem to me that this is the best solution. A few lined down the same method, the condition is checked for being defined, and if not, then no new filtering expression is added. I think the issue comes with the filtering operand reset which happens before the check is performed. If you move it inside the check, then there should be no error.

**Pull request merged**